### PR TITLE
Docs: Fix maven site generation

### PR DESCRIPTION
**Pull request by chingor13**
When we upgraded maven-javadoc-plugin to 3.0.1, we lost the configuration to aggregate the child javadocs - which caused the doc site to be missing all the javadoc.

This configuration was moved from a configuration option to a report.

This also removed unnecessary reporting configuration already specified in the reporting section.

### CODE REVIEW

File: google-cloud-clients/pom.xml [hunk 1]
```diff
@@ -700,90 +700,6 @@
...
-                  <group>
-                    <title>Deprecated packages</title>
-                    <packages>com.google.cloud.compute.deprecated*</packages>
-                  </group>
-                </groups>
-                <links>
-                  <link>https://developers.google.com/protocol-buffers/docs/reference/java/</link>
-                  <link>http://google.github.io/google-auth-library-java/releases/latest/apidocs/</link>
-                  <link>http://googleapis.github.io/gax-java/${gax.version}/apidocs/</link>
-                  <link>https://googleapis.github.io/api-common-java/${api.common.java.version}/apidocs/</link>
```
Comment (kolea2) [inline google-cloud-clients/pom.xml]: When making the change to add the api-common-java link, I didn't realize this was specified in two places. I'll create a PR for that.

Comment (kolea2) [reply google-cloud-clients/pom.xml]: #4236

**Comment by sduskis** [comment]
OSX java 8 fails: "[14:48:26][ERROR] Cannot check out non-mergeable pull request as if merged"

**Comment by chingor13** [comment]
CI needed to re-run, PTAL.

**Pull request merged**